sheepdog/utils/__init__.py

```diff
@@ -534,18 +529,17 @@ def lookup_program(psql_driver, program):
 
 def proxy_request(project_id, uuid, data, args, headers, method, action,
                   indexd_client, dry_run=False):
+    """Proxy a request to an object storage"""
+
     node = get_node(project_id, uuid)
     indexd_doc = indexd_client.get(uuid)
-
-    program, project = project_id.split('-', 1)
-    s3_url = generate_s3_url(
-        host=flask.current_app.config['SUBMISSION']['host'],
-        bucket=flask.current_app.config['SUBMISSION']['bucket'],
-        program=program,
-        project=project,
-        uuid=uuid,
-        file_name=indexd_doc.file_name,
-    )
+    s3_url = None
+    key_name = None
+    for url, metadata in indexd_doc.urls_metadata.items():
+        if metadata.get('type') == PRIMARY_URL_TYPE:
+            s3_url = url
+            key_name = '/'.join(s3_url.split('/')[1:])
```

Comment:
this won't be right, because urls are in form:
`s3://host/bucket/project_id/uuid/filename/`, so the `join(s3_url.split('/')[1:])` won't have the correct key. e.g.:

    >>> s3_url = 's3://cleversafe.service.consul/stage-submission-5/EXCEPTIONAL_RESPONDERS-ER/0beb9959-ea1b-4cc5-879b-494710449bcb'
    >>> s3_url.split('/')
    ['s3:', '', 'cleversafe.service.consul', 'stage-submission-5', 'EXCEPTIONAL_RESPONDERS-ER', '0beb9959-ea1b-4cc5-879b-494710449bcb']
    >>> '/'.join(s3_url.split('/'))
    's3://cleversafe.service.consul/stage-submission-5/EXCEPTIONAL_RESPONDERS-ER/0beb9959-ea1b-4cc5-879b-494710449bcb'
    >>> '/'.join(s3_url.split('/')[1:])
    '/cleversafe.service.consul/stage-submission-5/EXCEPTIONAL_RESPONDERS-ER/0beb9959-ea1b-4cc5-879b-494710449bcb'

Reply:
If I am not mistaken, key is everything except for the `host` part.

Reply:
something like this might work?

    from urlparse import urlparse
    parsed = urlparse(s3_url)
    key = '/'.join(parsed.path.strip('/').split('/')

Reply:
Yep you are correct. I am making changes to this currently. I'll re-title this as WIP until I push my changes.